lib/bolt/module_installer.rb

```diff
@@ -49,28 +49,28 @@ def add(name, specs, puppetfile_path, moduledir, config_path)
 
       begin
         resolve_specs.add_specs('name' => name)
-        puppetfile = Resolver.new.resolve(resolve_specs)
+        puppetfile = Resolver.new.resolve(resolve_specs, config)
       rescue Bolt::Error
         project_specs.add_specs('name' => name)
-        puppetfile = Resolver.new.resolve(project_specs)
+        puppetfile = Resolver.new.resolve(project_specs, config)
```

Comment:
I think we also want to catch if this fails and install module as if the user had passed `--no-resolve`

Reply:
I'm not sure what you mean... `bolt module add` doesn't support `--no-resolve`, and it doesn't really make sense to me to support loading modules from the Puppetfile and adding a new module to it. If you're using `--no-resolve` to install modules, you should be adding new modules manually.

Reply:
Can we catch an error here then and, if `module-install` is configured recommend the manual workflow?

Reply:
While config is currently being passed to `Bolt::ModuleInstaller::Resolver`, it doesn't actually get passed to `puppetfile-resolver`. If resolution is failing here, it's because the modules cannot be resolved at all. I don't think we should be recommending the manual workflow in that case.

Reply:
But the resolution might fail because users have a module published to their alternate Forge (that they've configured) that isn't available on the public Forge, which can cause resolution to fail. In that case users can manually add the module to their Puppetfile and still use `module install --no-resolve` to install from the correct place, even though resolution will fail.

Reply:
I'm not particularly keen on falling back to `--no-resolve` behavior if resolution fails, since I think it results in somewhat unintuitive behavior. I think there are two approaches we could take instead:

- Only pass `module-install` config if a user runs `module install --no-resolve`. Otherwise, warn that the config is not used and link to the known issue. Since we currently _only_ support this config for installing modules, not resolving, restricting the config to only being used when using the `--no-resolve` flag makes sense. This also puts users who need this config in the habit of running `--no-resolve` instead of expecting Bolt to try and resolve modules and potentially succeed with a different set of modules than they're expecting.

- Immediately fall back to `--no-resolve` behavior for `module install` if `module-install` is configured. This way Bolt won't even attempt to resolve modules. We can add a warning here as well that Bolt is skipping resolution because `module-install` is configured, with a link to the known issue. This allows users to still run `module install` to set up a project they downloaded, though it does still add what I think is slightly unintuitive behavior to the command for project authors.

Reply:
I think the best thing to do for both `module add` (here) and `module install` (below) is to try to resolve, catch errors and print a message telling the user they can try manually adding the module to the Puppetfile and installing with `--no-resolve` if they have `module-install.forge.baseurl` configured.